luminance-gl/src/gl33/state.rs

```diff
@@ -132,18 +165,18 @@ impl GLState {
   fn get_from_context() -> Result<Self, StateQueryError> {
     unsafe {
       let binding_stack = BindingStack::new();
-      let viewport = get_ctx_viewport()?;
-      let clear_color = get_ctx_clear_color()?;
-      let blending_state = get_ctx_blending_state()?;
-      let blending_equation = get_ctx_blending_equation()?;
-      let blending_func = get_ctx_blending_factors()?;
-      let depth_test = get_ctx_depth_test()?;
-      let depth_test_comparison = DepthComparison::Less;
-      let face_culling_state = get_ctx_face_culling_state()?;
-      let face_culling_order = get_ctx_face_culling_order()?;
-      let face_culling_mode = get_ctx_face_culling_mode()?;
-      let vertex_restart = get_ctx_vertex_restart()?;
-      let patch_vertex_nb = 0;
+      let viewport = CachedValue::new(get_ctx_viewport()?);
+      let clear_color = CachedValue::new(get_ctx_clear_color()?);
+      let blending_state = CachedValue::new(get_ctx_blending_state()?);
+      let blending_equation = CachedValue::new(get_ctx_blending_equation()?);
+      let blending_func = CachedValue::new(get_ctx_blending_factors()?);
+      let depth_test = CachedValue::new(get_ctx_depth_test()?);
+      let depth_test_comparison = CachedValue::new(DepthComparison::Less);
+      let face_culling_state = CachedValue::new(get_ctx_face_culling_state()?);
+      let face_culling_order = CachedValue::new(get_ctx_face_culling_order()?);
+      let face_culling_mode = CachedValue::new(get_ctx_face_culling_mode()?);
+      let vertex_restart = CachedValue::new(get_ctx_vertex_restart()?);
+      let patch_vertex_nb = CachedValue::new(0);
       let current_texture_unit = get_ctx_current_texture_unit()?;
```

Comment:
So those don’t get cached?

Reply:
Sorry, maybe I understood it wrong, but they do get cached, because the state is stored and compared on change. Or did you mean something else?

Reply:
Yeah it’s on me, sorry.